### ISSUE TITLE
dhcpv6 has log: NDISC: Error sending Router Solicitation: Network is unreachable

### ISSUE THREAD

**Issue by lvchangsheng** [systemd/systemd#29686]
### systemd version the issue has been seen with

234

### CPU architectures issue was seen on

None

### Expected behaviour you didn't see
```
cat /etc/systemd/network/10-eth0.network
# Copyright 2022-present Meta Platforms, Inc. All Rights Reserved.
#
# This program file is free software; you can redistribute it and/or modify it
# under the terms of the GNU General Public License as published by the
# Free Software Foundation; version 2 of the License.
#
# This program is distributed in the hope that it will be useful, but WITHOUT
# ANY WARRANTY; without even the implied warranty of MERCHANTABILITY or
# FITNESS FOR A PARTICULAR PURPOSE.  See the GNU General Public License
# for more details.
#
# You should have received a copy of the GNU General Public License
# along with this program in a file named COPYING; if not, write to the
# Free Software Foundation, Inc.,
# 51 Franklin Street, Fifth Floor,
# Boston, MA 02110-1301 USA
#
[Match]
Name=eth0

[Network]
IPv6AcceptRA=true
DHCP=yes
```

### Unexpected behaviour you saw

i use to dhcpv6 for address, but it has "NDISC: Error sending Router Solicitation: Network is unreachable"
there not ipv6 address , why.
```
root@bmc:~# systemctl --version
systemd 234
-PAM -AUDIT -SELINUX +IMA -APPARMOR +SMACK +SYSVINIT +UTMP -LIBCRYPTSETUP -GCRYPT -GNUTLS +ACL +XZ -LZ4 -SECCOMP +BLKID -ELFUTILS +KMOD -IDN2 -IDN default-hierarchy=hybrid
root@bmc:~#

root@bmc:~# ifconfig
eth0      Link encap:Ethernet  HWaddr 58:69:6C:FB:20:29
          inet addr:10.110.181.42  Bcast:10.110.181.255  Mask:255.255.255.0
          inet6 addr: fe80::5a69:6cff:fefb:2029/64 Scope:Link
          UP BROADCAST RUNNING MULTICAST  MTU:1500  Metric:1
          RX packets:4906 errors:0 dropped:0 overruns:0 frame:0
          TX packets:1428 errors:0 dropped:0 overruns:0 carrier:0
          collisions:0 txqueuelen:1000
          RX bytes:972765 (949.9 KiB)  TX bytes:195412 (190.8 KiB)
```

### Additional program output to the terminal or log subsystem illustrating the issue

```sh
root@bmc:~# systemctl stop systemd-networkd
_LOG_LEVEL=debug /usr/lib/systemd/systemd-networkdWarning: Stopping systemd-networkd.service, but it can still be activated by:
  systemd-networkd.socket
root@bmc:~# SYSTEMD_LOG_LEVEL=debug /usr/lib/systemd/systemd-networkd
-bash: /usr/lib/systemd/systemd-networkd: No such file or directory
root@bmc:~#
root@bmc:~#
root@bmc:~# systemctl stop systemd-networkd
EL=debug /lib/systemd/systemd-networkdWarning: Stopping systemd-networkd.service, but it can still be activated by:
  systemd-networkd.socket
root@bmc:~# SYSTEMD_LOG_LEVEL=debug /lib/systemd/systemd-networkd
Failed to read $container of PID 1, ignoring: Permission denied
Found container virtualization none.
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=Hello cookie=1 reply_cookie=0 error=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.5 object=n/a interface=n/a member=n/a cookie=1 reply_cookie=1 error=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=AddMatch cookie=2 reply_cookie=0 error=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.5 object=n/a interface=n/a member=n/a cookie=3 reply_cookie=2 error=n/a
Sent message type=method_call sender=n/a destination=org.freedesktop.DBus object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=RequestName cookie=3 reply_cookie=0 error=n/a
Got message type=method_return sender=org.freedesktop.DBus destination=:1.5 object=n/a interface=n/a member=n/a cookie=5 reply_cookie=3 error=n/a
Failed to open configuration file '/etc/systemd/networkd.conf': No such file or directory
timestamp of '/etc/systemd/network' changed
eth0.4088: Flags change: +UP +LOWER_UP +RUNNING +MULTICAST +BROADCAST
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_35 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=4 reply_cookie=0 error=n/a
eth0.4088: Link 5 added
eth0.4088: udev initialized link
eth0.4088: Saved original MTU: 1500
sit0: MAC address not found for new device, continuing without
sit0: Flags change: +NOARP
sit0: Link 4 added
sit0: udev initialized link
sit0: Saved original MTU: 1480
usb0: Flags change: +MULTICAST +BROADCAST
usb0: Link 3 added
usb0: udev initialized link
usb0: Saved original MTU: 1500
eth0: Flags change: +UP +LOWER_UP +RUNNING +MULTICAST +BROADCAST
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=5 reply_cookie=0 error=n/a
eth0: Link 2 added
eth0: udev initialized link
eth0: Saved original MTU: 1500
lo: Flags change: +LOOPBACK +UP +LOWER_UP +RUNNING
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_31 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=6 reply_cookie=0 error=n/a
lo: Link 1 added
lo: udev initialized link
lo: Saved original MTU: 0
eth0.4088: Adding address: fe80::5a69:6cff:fefb:2029/64 (valid forever)
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_35 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=7 reply_cookie=0 error=n/a
eth0.4088: Gained IPv6LL
eth0: Adding address: fe80::5a69:6cff:fefb:2029/64 (valid forever)
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=8 reply_cookie=0 error=n/a
eth0: Gained IPv6LL
lo: Adding address: ::1/128 (valid forever)
eth0.4088: Adding address: 240.1.1.1/30 (valid forever)
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_35 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=9 reply_cookie=0 error=n/a
eth0: Adding address: 10.110.181.42/24 (valid for 23h 59min 48s)
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=10 reply_cookie=0 error=n/a
lo: Adding address: 127.0.0.1/8 (valid forever)
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=11 reply_cookie=0 error=n/a
Enumeration completed
eth0.4088: Link state is up-to-date
No virtualization found in DMI
No virtualization found in CPUID
Virtualization XEN not found, /proc/xen/capabilities does not exist
/proc/device-tree: No such file or directory
No virtualization found in /proc/cpuinfo.
This platform does not support /proc/sysinfo
Found VM virtualization none
eth0.4088: Unmanaged
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_35 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=12 reply_cookie=0 error=n/a
sit0: Link state is up-to-date
sit0: Unmanaged
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_34 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=13 reply_cookie=0 error=n/a
usb0: Link state is up-to-date
usb0: Unmanaged
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_33 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=14 reply_cookie=0 error=n/a
eth0: Link state is up-to-date
eth0: found matching network '/etc/systemd/network/10-eth0.network'
LLDP: Started LLDP client
eth0: Started LLDP.
eth0: Acquiring DHCPv4 lease
DHCP CLIENT (0x683f248d): STARTED on ifindex 2
DHCPv6 CLIENT: Started in Information request mode
eth0: Acquiring DHCPv6 lease
eth0: Discovering IPv6 routers
NDISC: Started IPv6 Router Solicitation client
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=15 reply_cookie=0 error=n/a
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=16 reply_cookie=0 error=n/a
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=17 reply_cookie=0 error=n/a
NDISC: Error sending Router Solicitation: Network is unreachable
NDISC: Stopping IPv6 Router Solicitation client
DHCP CLIENT (0x683f248d): DISCOVER
DHCPv6 CLIENT: Next retransmission in 1s
DHCP CLIENT (0x683f248d): OFFER
DHCP CLIENT (0x683f248d): REQUEST (requesting)
lo: Link state is up-to-date
lo: Unmanaged
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_31 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=18 reply_cookie=0 error=n/a
DHCP CLIENT (0x683f248d): ACK
DHCP CLIENT (0x683f248d): lease expires in 23h 59min 57s
DHCP CLIENT (0x683f248d): T2 expires in 20h 59min 57s
DHCP CLIENT (0x683f248d): T1 expires in 11h 59min 57s
eth0: DHCPv4 address 10.110.181.42/24 via 10.110.181.1
eth0: Removing address: 10.110.181.42/24 (valid for 23h 59min 44s)
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=19 reply_cookie=0 error=n/a
eth0: Adding address: 10.110.181.42/24 (valid for 1d)
Sent message type=signal sender=n/a destination=n/a object=/org/freedesktop/network1/link/_32 interface=org.freedesktop.DBus.Properties member=PropertiesChanged cookie=20 reply_cookie=0 error=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.5 object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=2 reply_cookie=0 error=n/a
Got message type=signal sender=org.freedesktop.DBus destination=:1.5 object=/org/freedesktop/DBus interface=org.freedesktop.DBus member=NameAcquired cookie=4 reply_cookie=0 error=n/a
DHCPv6 CLIENT: Next retransmission in 1s
DHCPv6 CLIENT: Next retransmission in 4s
DHCPv6 CLIENT: Next retransmission in 7s
DHCPv6 CLIENT: Next retransmission in 16s
LLDP: Invoking callback for 'a'.
LLDP: Successfully processed LLDP datagram.
LLDP: Invoking callback for 'a'.
LLDP: Successfully processed LLDP datagram.
```

**Comment by yuwata** [comment]
Already fixed by 852bf93826b151be8b85d894071d95eb3b0d4498. Please use newer version, or backport the commit.

**Comment by lvchangsheng** [comment]
why my device not have dhcpv6 ipv6 address，if there same other need config
